pandas/_libs/tslibs/period.pyx

```diff
@@ -2397,11 +2402,14 @@ class Period(_Period):
     hour : int, default 0
     minute : int, default 0
     second : int, default 0
+    dayfirst : bool, default None
+    yearfirst : bool, default None
     """
 
     def __new__(cls, value=None, freq=None, ordinal=None,
                 year=None, month=None, quarter=None, day=None,
-                hour=None, minute=None, second=None):
+                hour=None, minute=None, second=None, dayfirst=None,
```

Comment:
@jbrockmendel @TomAugspurger any problem adding this ?

Reply:
@qwhelan can you add a versionadded in the doc-string (and add an expl here) or the new params

Reply:
Fine by me.  Elsewhere we're trying to move towards fewer arguments in the constructors.  Should we consider a dedicated constructor for this?  (not necessarily for this PR)

Reply:
@jbrockmendel I'm indifferent to the form this takes - but it would seem that option would largely entail renaming the current `__new__()` and adding a new `__new__()` that simply does not expose these parameters. Is that what you have in mind?